pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -214,6 +242,83 @@ func (r *Reconciler) createReceiveAdapter(ctx context.Context, src *v1alpha1.Api
 	return ra, err
 }
 
+func (r *Reconciler) createEventTypes(ctx context.Context, src *v1alpha1.ApiServerSource) error {
+	current, err := r.getEventTypes(ctx, src)
+	if err != nil {
+		logging.FromContext(ctx).Error("Unable to get existing event types", zap.Error(err))
+		return err
+	}
+
+	expected, err := r.makeEventTypes(src)
+	if err != nil {
+		return err
+	}
+
+	missing := r.computeDiff(current, expected)
+	for _, eventType := range missing {
+		if _, err = r.EventingClientSet.EventingV1alpha1().EventTypes(src.Namespace).Create(&eventType); err != nil {
+			return err
+		}
+		logging.FromContext(ctx).Desugar().Info("EventType created", zap.Any("eventType", eventType))
+	}
+	return err
+}
+
+func (r *Reconciler) getEventTypes(ctx context.Context, src *v1alpha1.ApiServerSource) ([]eventingv1alpha1.EventType, error) {
+	etl, err := r.EventingClientSet.EventingV1alpha1().EventTypes(src.Namespace).List(metav1.ListOptions{
+		LabelSelector: r.getLabelSelector(src).String(),
+	})
+	if err != nil {
+		logging.FromContext(ctx).Desugar().Error("Unable to list event types: %v", zap.Error(err))
+		return nil, err
+	}
+	eventTypes := make([]eventingv1alpha1.EventType, 0)
+	for _, et := range etl.Items {
+		if metav1.IsControlledBy(&et, src) {
+			eventTypes = append(eventTypes, et)
+		}
+	}
+	return eventTypes, nil
+}
+
+func (r *Reconciler) makeEventTypes(src *v1alpha1.ApiServerSource) ([]eventingv1alpha1.EventType, error) {
+	eventTypes := make([]eventingv1alpha1.EventType, 0)
+	args := &resources.EventTypeArgs{
+		Src:    src,
+		Source: "ScottMagic",
```

Comment:
@n3wscott how to properly populate the EventType source?

Reply:
You only really know at Pod runtime.

Reply:
And it could change.

Reply:
Done the same thing as in the adapter. Not optimal, but is what we have for now.